wallet-new/src/Cardano/Wallet/API/Request/Sort.hs

```diff
@@ -183,6 +178,47 @@ instance ( HasServer subApi ctx
 
         in route (Proxy :: Proxy subApi) context delayed
 
+-- | Logs all non-identity sorting parameters.
+--
+-- Example: @filter: asc id, desc balance@.
+--
+-- Default implementation of this instance for 'SortBy params res'
+-- (note OVERLAPPING) would print logs in format of
+-- "<some name dependant on 'syms'>: <prettified SortOperations>".
+-- With current implementation of 'SortOperations' the best logs we can get
+-- would be like "filter: asc, desc, none",
+-- or "filter (id, created_at, balance): asc, desc, none",
+-- because 'SortOperations' itself doesn't remember name of parameter it sorts
```

Comment:
It does, actually :) In the GADT constructor, we stuff a bunch of constraints. One of them is `KnownSymbol (IndexToQueryParam res ix)`, so we can unpack the `SortOp` constructor and access that type:

```haskell
foo :: SortOperations a -> [String]
foo NoSorts = []
foo (SortOp (sop :: SortOperation ix a) rest) =
  symbolVal (Proxy :: Proxy (IndexToQueryParam a ix)) : foo rest
```

Extracts the list of values that were passed. Eg:

```haskell
"?sort_by=id&sort_by=ASC[timestamp]"
-- parses into
parsed = SortByIndex SortDesc (Proxy :: Proxy WalletId)
  `SortOp` SortByIndex SortAsc (Proxy :: Proxy (V1 Core.Timestamp))
  `SortOp` NoSorts

>>> print (foo parsed)
["id", "timestamp"]
```

Reply:
Oh, good point.
Then I agree with all points above, perhaps it can be implemented without `OVERLAPPING`, just defining `instance ApiHasArgClass` with `apiArgName = "sort"` and `instance ApiCanLogArg` with `toLogParamInfo` printing parameter names and corresponding values, similarly to current logic here https://goo.gl/AvytVJ - or something like that.
And same thing for filtering. Should be cool eventually :)

Reply:
I believe this comment _was_ true before I rebased on top the latest work done by @parsonsmatt which actually changes the way `FilterBy` and `SortBy` were declared (from `[Symbol]` to `[*]`), allowing more information to be embedded at the type-level.

Reply:
I say :tada: then ))